Source/WebCore/rendering/RenderListItem.cpp

```diff
@@ -231,25 +231,47 @@ void RenderListItem::updateValueNow() const
         }
     }
 
+    int defaultIncrement = (orderedList && orderedList->isReversed()) ? -1 : 1;
+    // FIXME: This should account for counter-reset as well.
```

Comment:
There are some tests on WPT, but also our own layout test seems to exercise this (although the expected results are incorrect). I'll make a basic fix so that test passes, but I'll do the full support in a follow up.